dynamic_library/src/posix/dynamic_library.cc

```diff
@@ -0,0 +1,102 @@
+#include <leatherman/dynamic_library/dynamic_library.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <dlfcn.h>
+
+using namespace std;
+
+namespace leatherman { namespace dynamic_library {
+
+    DynamicLibrary DynamicLibrary::find_by_pattern(std::string const& pattern)
+    {
+        // POSIX doesn't have this capability
+        return DynamicLibrary();
+    }
+
+    DynamicLibrary DynamicLibrary::find_by_symbol(std::string const& symbol)
+    {
+        DynamicLibrary library;
+
+        // Load the "null" library; this will cause dlsym to search for the symbol
+        void* handle = dlopen(nullptr, RTLD_GLOBAL | RTLD_LAZY);
+        if (!handle) {
+            return library;
+        }
+
+        // Check to see if a search for the symbol succeeds
+        if (!dlsym(handle, symbol.c_str())) {
+            dlclose(handle);
+            return library;
+        }
+
+        // At least one loaded module will resolve the given symbol
+        // Return this handle to allow the caller to search for other symbols
+        library._handle = handle;
+        library._first_load = false;
+        return library;
+    }
+
+    bool DynamicLibrary::load(string const& name)
+    {
+        close();
+
+        // Don't actually perform a load to determine if it is already loaded
+#ifdef _AIX
```

Comment:
Need to update based on https://github.com/puppetlabs/facter/commit/858e51a39948a383e0e651d0ca6d3fc1b17edf41.